noir_stdlib/src/collections/map.nr

```diff
@@ -104,30 +105,146 @@ impl<K, V, N, B, H> HashMap<K, V, N, B> {
         self._len == 0
     }
 
-    // Returns a vector of all active keys.
-    pub fn keys(self) -> Vec<K>{
-        let mut keys = Vec::new();
+    // Returns array of all active key-value pairs.
+    // Array size is restricted to N because of 
+    // compile time limitations and nested slices temporary ban, 
+    // hovewer first len() positioned with pairs 
+    // and afterwards all elements are zeroed().
+    pub fn iter(self) -> [(K, V); N]{
```

Comment:
I though about using `Option`, but decided that it will bring performance overhead compared to pure tuples, especially since we are sure about each elements `is_some` (`is_none`) prematurely. Though it is indeed more safe to have homogeneous `Option` wrapped instead of `zeroed`.